Module-7-Assignment/Question-7/ManageClassController.cls

```diff
@@ -0,0 +1,55 @@
+/**
+ * Controller for page ManageClass.
+ */ 
+public class ManageClassController 
+{
+    public List<Class__c> classes {get;set;}
+    public Id classId {get;set;}
+    public String className {get;set;}
+    public String classDescription{get;set;}
+    public String classFee{get;set;}
+    public String classMaxSize{get;set;}
+    public boolean isEdit{get;set;}
+    
+    /**
+     * Constructor to initialize the isEdit variable to false and get all the classes.
+     */ 
+    public ManageClassController()
+    {
+        isEdit = false;
+        classes = new List<Class__c>();
+        classes = [SELECT Name FROM Class__c];
+    }
+    
+    /**
+     * This method sets the isEdit field to true in order to display the Edit Details Section
+     */ 
+    public void editClassDetails()
+    {
+        isEdit = true;  
+    }
+    
+    /**
+     * This method saves new data of the class and redirects to the same page.
+     */ 
+    public PageReference SaveDetails()
+    {
+        Class__c currentClass = [SELECT Name, DetailedDescription__c , Currency__c, MaxSize__c FROM Class__c WHERE Id =: classId];
+        currentClass.Name = className;
+        currentClass.DetailedDescription__c = classDescription;
+        currentClass.Currency__c = !String.isBlank(classFee) ? Decimal.valueOf(classFee) : 0;
+        currentClass.MaxSize__c = !String.isBlank(classMaxSize) ? Decimal.valueOf(classMaxSize) : 0;
+        update currentClass;
+        return (new PageReference('https://c.ap15.visual.force.com/apex/manageClass?core.apexpages.request.devconsole=1'));
```

Comment:
This is a hard coded redirect. This will not work on other orgs

Reply:
Done